curator-x-discovery/src/main/java/org/apache/curator/x/discovery/details/ServiceDiscoveryImpl.java

```diff
@@ -207,9 +207,15 @@ protected void     internalRegisterService(ServiceInstance<T> service) throws Ex
         {
             try
             {
-                CreateMode      mode = (service.getServiceType() == ServiceType.DYNAMIC) ? CreateMode.EPHEMERAL : CreateMode.PERSISTENT;
-                client.create().creatingParentsIfNeeded().withMode(mode).forPath(path, bytes);
-                isDone = true;
+                synchronized(service) {
+                    // Check that the service didn't get deleted while we were reconnecting
+                    if (services.containsKey(service.getId()))
```

Comment:
Why is this needed? The test passes without this and logically it seems unnecessary.